src/network/networkctl.c

```diff
@@ -2919,6 +2928,270 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
+        if (r < 0)
+                return log_oom();
+
+        return edit_files_add(context, dropin_path, old_dropin, comment_paths);
+}
+
+static int verb_edit(int argc, char *argv[], void *userdata) {
+        _cleanup_(edit_file_context_done) EditFileContext context = {
+                .marker_start = DROPIN_MARKER_START,
+                .marker_end = DROPIN_MARKER_END,
+                .remove_parent = !!arg_drop_in,
+        };
+        _cleanup_free_ char *drop_in = NULL;
+        bool networkd_reload = false, udevd_reload = false;
+        int r;
+
+        if (!on_tty())
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit network config files if not on a tty.");
+
+        r = mac_selinux_init();
+        if (r < 0)
+                return r;
+
+        r = get_arg_drop_in(&drop_in);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, strv_skip(argv, 1)) {
+                _cleanup_strv_free_ char **dropins = NULL;
+                _cleanup_free_ char *path = NULL;
+
+                if (ENDSWITH_SET(*name, ".network", ".netdev"))
+                        networkd_reload = true;
+                else if (endswith(*name, ".link"))
+                        udevd_reload = true;
+                else
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid network config name '%s'.", *name);
+
+                r = get_config_files_by_name(*name, &path, &dropins);
+                if (r == -ENOENT) {
+                        if (drop_in)
+                                return log_error_errno(r, "Cannot find network config '%s'.", *name);
+
+                        log_debug("No existing network config '%s' found, creating a new file.", *name);
+
+                        path = path_join(NETWORK_DIRS[0], *name);
+                        if (!path)
+                                return log_oom();
+
+                        r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                        continue;
+                }
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get the path of network config '%s': %m", *name);
+
+                r = add_config_to_edit(&context, path, drop_in, dropins);
+                if (r < 0)
+                        return r;
+        }
+
+        r = do_edit_files_and_install(&context);
+        if (r < 0)
+                return r;
+
+        if (arg_no_reload)
+                return 0;
+
+        if (udevd_reload) {
+                r = safe_fork("(udevadm)",
+                              FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG|FORK_WAIT|FORK_FLUSH_STDIO,
+                              NULL);
+                if (r < 0)
+                        return r;
+                if (r == 0) { /* Child */
+                        execl("/usr/bin/udevadm", "udevadm", "control", "--reload", NULL);
```

Comment:
instead of shelling out I'd prefer to issue the equivalent of "systemctl reload systemd-udevd.service" via the bus

Reply:
actually, nah, there's no point in reloading udev config. as that's not when the .link files are read. Instead, the .link files are loaded when a network interface is actually processed by udev. Hence what you want here is to trigger all network devices, and wait for the event to propagate through the thing.

i.e. the equivalent of `udevadm trigger -s net -w`

Reply:
Will shelling out be fine for retriggering udev?

Reply:
After the change the CI is failing:

> [   57.486760] testsuite-74.sh[1681]: FAIL: 'alias test_alias' not found in:
> [   57.487406] testsuite-74.sh[1681]: 2: test2: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000

I think reload is still necessary 🤔

Reply:
> I think reload is still necessary

[systemd.link(5)](https://github.com/systemd/systemd/blob/94d82b59807488343bd206ba8f111d24ff17d9c1/man/systemd.link.xml#L1149) also says so.